src/component/todo-item/todo-item.ts

```diff
@@ -0,0 +1,69 @@
+import { todoStore } from '../../utils/todo-store';
+import { Todo } from '../../models/todo';
+export class TodoItem extends HTMLElement {
+  private todoStore = todoStore;
+  private checkbox: HTMLInputElement | null = null;
+
+  constructor() {
+    super();
+  }
+
+  connectedCallback(): void {
+    this.render();
+    this.checkbox = this.querySelector('input');
+    this.addEventListeners();
+  }
+
+  disconnectedCallback(): void {
+    this.removeEventListeners();
+  }
+
+  private render(): void {
+    this.classList.add('todo-item');
+    const text = this.getAttribute('text') || '';
+    const done = this.getAttribute('done') === 'true';
+
+    this.innerHTML = `
+      <style>
+        :host {
+          display: block;
+          width: 100%;
+        }
+        li {
+          list-style: none;
+          padding: 10px;
+          border-bottom: 1px solid #ddd;
+        }
+        input:checked + span {
+          text-decoration: line-through;
+          color: #999;
+        }
+      </style>
+      <li>
+        <input type="checkbox" ${done ? 'checked' : ''}>
+        <span>${text}</span>
+      </li>
+    `;
+  }
+
+  private addEventListeners(): void {
+    this.checkbox?.addEventListener('change', this.onChange);
+  }
+
+  private removeEventListeners(): void {
+    this.checkbox?.removeEventListener('change', this.onChange);
+  }
+
+  private onChange = (e: Event) => this.handleChange(e);
+
+  private handleChange(event: Event): void {
+    const target = event.target as HTMLInputElement;
+    const todo = this.todoStore.getState().find((t: Todo) => String(t.id) === this.id);
+    if (todo) {
+      todo.done = target.checked;
+      this.todoStore.setState(this.todoStore.getState());
```

Comment:
여기에서 gestState 값을 setState 해주는게 storeState 내부에 있는 state는 이미 변경되어서 그런 걸까요 ?
완전히 분리해서 관리 -> lodash 사용하여 깊은 복사 해주는게 좋을 것 같습니다.

```suggestion
     const storeTodos = cloneDeep(this.todoStore.getState());
    const todo = storeTodos.find((t: Todo) => String(t.id) === this.id);

    if (todo) {
      todo.done = target.checked;
      this.todoStore.setState(storeTodos);
    }
```

Reply:
깊은 복사정도는 직접 구현을... (ㅌㅌ)